portal/utils.py

```diff
@@ -1,6 +1,110 @@
+from math import ceil
+
+from config import POINTS_FOR_PARTY_MATCH, POINTS_FOR_INTEREST_MATCH, EXPECTED_CHILDREN
+from portal.models import Student, Family
+
+
 def get_invalid_id_popup():
     return {'message': "The ID provided is invalid.", 'state': 'danger'}
 
 
 def get_student_does_not_exist_popup():
     return {'message': "Failed to find student.", 'state': 'danger'}
+
+
+def create_families_from_parents():
+    if Student.objects.filter(child=False).count() % 2 != 0:
+        to_remove = Student.objects.filter(child=False, partner__isnull=True).last()
+        # ToDO(martinzlocha): Inform student that he cant be a parent because there is an odd number and he was last
+        to_remove.delete()
+
+    __pair_up_parents()
+
+    paired = list(Student.objects.filter(child=False, partner__isnull=False, confirmed=True))
+
+    for i in range(int(len(paired) / 2)):
```

Comment:
len(paired) will always be even because if it isn't then one person is removed from the list in line 19